zerver/views/user_settings.py

```diff
@@ -168,14 +168,18 @@ def json_set_avatar(request, user_profile):
         return json_error("You must upload exactly one avatar.")
 
     user_file = list(request.FILES.values())[0]
-    upload_avatar_image(user_file, user_profile, user_profile.email)
-    do_change_avatar_source(user_profile, UserProfile.AVATAR_FROM_USER)
-    user_avatar_url = avatar_url(user_profile)
-
-    json_result = dict(
-        avatar_url = user_avatar_url
-    )
-    return json_success(json_result)
+    try:
+        upload_avatar_image(user_file, user_profile, user_profile.email)
+        do_change_avatar_source(user_profile, UserProfile.AVATAR_FROM_USER)
+        user_avatar_url = avatar_url(user_profile)
+
+        json_result = dict(
+            avatar_url = user_avatar_url
+        )
+        return json_success(json_result)
+    except IOError as e:
+        print(e)
+        return json_error("Could not decode image")
```

Comment:
This error should probably provide more feedback to the user about what they probably did wrong, since it's user-facing.  E.g. "Could not decode avatar image; did you upload an image file?"